pkg/sampling/doc.go

```diff
@@ -0,0 +1,9 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package sampling hanldes probability sampling information in the W3C tracestate.
+package sampling // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/sampling"
+
+// Note that the following does not work for non-component packages:
```

Comment:
Is there an issue with the go:generate instruction here?

Reply:
Yes. All the packages in `pkg/*` do this, the generate step does not work.

Reply:
I left the comment-in because I was following the process documented for new components in CONTRIBUTING.md. I figured since this didn't work for non-component packages, I should leave a comment. I'm happy to remove this comment if you think I should.